### ISSUE TITLE
Blazor WASM significant slower load time in MS Edge Dev

### ISSUE THREAD

**Issue by vallgrenerik** [dotnet/aspnetcore#51670]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I don't know if this is a Blazor bug, or a Edge bug. But to inform, I create this issue here:
Opening a Blazor WASM site eg. https://www.erikvallgren.se (dotnet 6 default WASM template) inside Edge Dev browser, the load time is much slower than regular Edge.  
![image](https://github.com/dotnet/aspnetcore/assets/26163937/31d64208-6951-4042-bb66-4b15d2eb3039)

Is there something in the Blazor framework causing this, or is this a Edge "bug"? 

### Expected Behavior

The site to load faster

### Steps To Reproduce

Head to https://www.erikvallgren.se/ or https://mudblazor.com/ in both Edge Dev & Edge regular and see the difference.

### .NET Version

dotnet 6.0.7

**Comment by javiercn** [comment]
@vallgrenerik thanks for contacting us.

You are likely hitting https://learn.microsoft.com/en-us/deployedge/microsoft-edge-security-browse-safer

/cc: @danroth27

**Comment by vallgrenerik** [comment]
@javiercn 
Interesting, the settings did diff. On Edge Dev the "Enhance your security on the web" were enabled, and disabled on Edge (proper).
When disabled "Enhance your security on the web" in Edge Dev, the slow result was still there, so no improvement unfortunately.

**Comment by danroth27** [comment]
> When I disabled "Enhance your security on the web" in Edge Dev, the slow result was still there, so no improvement unfortunately.

@vallgrenerik Is there any chance that Edge Enhanced Security Mode (ESM) somehow got flipped back on even though you turned it off? Can you screen shot the setting for us in Edge Dev just so we can confirm?

**Comment by vallgrenerik** [comment]
Sure @danroth27 
I'm using private tabs, don't know if that has any impact. I don't think so... 😊
Here is some screenshots:
![image](https://github.com/dotnet/aspnetcore/assets/26163937/b1e436fe-6d3f-49ae-8f90-c5e5e8c43ad9)
![image](https://github.com/dotnet/aspnetcore/assets/26163937/db27213a-f28e-4028-a419-c78272aaf046)

**Comment by danroth27** [comment]
@vallgrenerik Thanks for confirming! I'm working with the Edge team to figure out what's going on here.

Can you also try with Edge Canary and Chrome Dev to see if you hit the same issue?

**Comment by vallgrenerik** [comment]
@danroth27 Sorry for the delay! 

I have now tried in Canary as well, and now the normal loading time is back.
Even in Edge Dev. I can't figure out what happened. 
Edge Dev is using the same version as when I had a slow loading time.   

One interesting thing I noticed when installing Edge Canary, is that Canary wouldn't install if I didn't close MS Teams. I tried multiple times to open the installation file but it just closed after a couple of seconds. 

Then I opened Task manager to see if any suspicious process were hanging, but no. When I searched in Task manager for processes containing "Edge" there were multiple "Edge update"-something-something, processes inside the MS Teams process. When I killed MS Teams, the setup of Canary started up, automatically. 
After that, all instances of Edge (Stable, Dev, Canary) have the same loading time (~600 ms) for a default .NET 6 WASM Blazor template. 

Sorry for the inconvenience, the problem must've been on my machine somehow.
